scss/_patterns_lists.scss

```diff
@@ -199,6 +220,84 @@ $list-step-bullet-margin: $sph--x-large;
   }
 }
 
+// Sets horizontal list section grid column count and makes list item borders span the full grid width.
+// $columns: Number of columns that the list items should be divided into.
+// $skip-cols: Number of columns at the start of each row to skip
+@mixin list-horizontal-section-row-width($columns, $skip-cols: 0) {
+  --columns: #{$columns + $skip-cols};
+  // List item borders span full grid width, not list item width.
+  // Multiply 100% width by the number of columns to create a border that spans the full grid width.
+  .p-list__item:nth-child(#{$columns}n + 1)::after {
+    background: $colors--theme--border-low-contrast;
+    content: '';
+    height: 1px;
+    left: 0;
+    position: absolute;
+    top: 0;
+    width: calc(100% * #{$columns} + #{$columns - 1} * map-get($grid-gutter-widths, default));
```

Comment:
It would be worth adding a comment line for this calculation explaning what it does and why. Math may be simple enough, but after a while it's hard to remember what was the intention behind it.

Reply:
Added a comment